lib/ansible/plugins/loader.py

```diff
@@ -588,6 +587,9 @@ def get(self, name, *args, **kwargs):
 
         if not class_only:
             try:
+                # A plugin may need to use its _load_name in __init__ (for example, to use self.set_options or self.get_option).
+                # Update the object before instantiating the class in case this is true
+                self._update_object(obj, name, path)
```

Comment:
I left the _update_object call below for backwards compatibility in case an __init__ has a side effect.

Reply:
An alternative to this line would be to use class_only=True where our code uses cache_loader (and any other loaders that might need _load_name in `__init__`) so _update_object is called before instantiation. I picked this because it also fixes possible third party code using the plugin loader.